File: cs50x/2014/pset1/mario.c
/****************************************************************************
 * mario.c
 *
 * Printing One size of mario pyramide.
 * Usage        : ./mario
 * Description  : http://docs.cs50.net/problems/mario/more/mario.html
 * Testing      : check50 2014.fall.pset1.mario mario.c
 ***************************************************************************/

#include <stdio.h>
#include <cs50.h>

// declaring a function for printing.
void print( char CharToPrint, int Times );

int main( void ) {
    int n;

    do {
        printf("Height: ");
        n = GetInt();
    } while ( n < 0 || n > 23 );

    // Printing Pyramid.
    for ( int line = 1; line <= n; line++ ) {
        print( ' ',  n - line);
        print( '#',  line + 1);
        print( '\n',  1);
    }

    return 0;
}

void print( char CharToPrint, int Times ) {
    for ( int i = 0; i < Times; i++ ){
        printf( "%c", CharToPrint );
    }
}

File: HarvardX_CS50x/pset6/mario/less/mario.py
'''
 	mario.py

	Printing of custom Mario pyramide.

	Usage        : python3 mario.py
	Description  : http://docs.cs50.net/problems/mario/less/mario.html

'''
def ask():
	while True:
		HeightStr = input("Height:")
		Height 	  = 0;

		if HeightStr.isdigit() :
			Height = int(HeightStr)
		else :
			continue

		if Height not in range(1, 24):
			return ask()

		return Height

def main():

	# this long code used to
	# learn about inputs and analize
	# instead cs50 library.
	Height 	  = ask()

	for line in range(1, Height+1) :
		print( ( " " * ( Height - line) ) + ( "#" * ( line + 1 ) ) )

if __name__ == '__main__':
	main();
